include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -120,6 +133,31 @@ enable_if_t<is_spirv_type_v<Ptr_T>, T> atomicCompSwap(Ptr_T ptr, T comparator, T
     return spirv::atomicCompareExchange<T, Ptr_T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, spv::MemorySemanticsMaskNone, value, comparator);
 }
 
+/**
+ * GLSL std 450
+ */
+
+template<typename T>
+inline int findMSB(T val);
+
+//template<>
+//inline int findMSB<int32_t>(int32_t val) { return spirv::FindSMsb(val); }
+template<>
+inline int findMSB<uint32_t>(uint32_t val) { return spirv::FindUMsb(val); }
+template<>
+inline int findMSB<uint64_t>(uint64_t val)
+{
+    NBL_CONSTEXPR_STATIC uint64_t HighBitsMask = (~uint64_t(0)) << 32;
+    NBL_CONSTEXPR_STATIC uint64_t LowBitsMask = ~uint32_t(0);
+    uint32_t highBits = (val & HighBitsMask) >> 32;
+    uint32_t lowBits = val & LowBitsMask;
+
+    if (highBits == 0)
+        return spirv::FindUMsb(lowBits);
+    return spirv::FindUMsb(highBits) + 32;
+}
+
```

Comment:
the idea with the namespaces is as follows:
spirv you literally cannot use from Host/CPU
glsl is to get GLSL-like syntax
hlsl or similar is what you use to get Xplatform code

So it makes sense to have a `hlsl::findMSB`, `glsl:findMSB` and a `spirv::findMSB` at the same time.

But only `hlsl::findMSB` should be available in C++

P.S. uint64_t version of findMSB doesn't exist for GLSL, so it should only live in `hlsl::findMSB`